Automattic/wp-calypso
# Rework Quick Start flow to avoid use of SitePicker

Fixes https://github.com/Automattic/wp-calypso/issues/66799
Related https://href.li/?D86923-code

#### Proposed Changes

See https://github.com/Automattic/wp-calypso/issues/66799 for some background. 

Changes introduced:
1. Avoid showing the SitePicker as recommended in https://github.com/Automattic/wp-calypso/issues/66799
2. Change the wording and banner style as recommended in https://github.com/Automattic/wp-calypso/issues/66799
3. Simplify the switch statement in `<ConciergeBanner/>` - there are now just two cases: has an appointment scheduled or has sessions available (previously was three)
4. Change `<Banner/>` component style to avoid text wrapping in `<a/>` or `<button/>` elements
5. Introduce `<PurchaseListConciergeBanner/>` to make logic easier to test; added some tests
6. Show banner at `/purchases/subscriptions/:site` as well as exiting location `/me/purchases`. 

Note that the banner should only appear for `en` users until we have translations.

When the user is on SiteA and has an appointment scheduled for SiteB, they will see a banner about their upcoming appointment for SiteB. This was the existing logic which is still present. We should consider showing a banner offering to schedule an appointment for SiteA if they have one available. Please consider this during the review. 

**What the user sees when they have an appointment scheduled**:
![image](https://user-images.githubusercontent.com/6851384/187829756-c2cafe14-8ee9-432b-aafb-72e8ad58e228.png)

**What the user sees when they have sessions available**:
![image](https://user-images.githubusercontent.com/6851384/187829603-af5d4ab3-4a4b-4129-b518-e3afe47881ac.png)

#### Testing Instructions
1. Checkout this branch
2. Start your local Calypso dev environment
3. Go to the A8c Support User site, enter a username for a user with included sessions (see Phabricator Diff) and select `calypso.localhost` from the **environment** dropdown. This user has some included session
4. Start a local support session and navigate to `purchases/subscriptions/SITE_WITH_SESSION` and you should see the banner:
![image](https://user-images.githubusercontent.com/6851384/187864636-201b2296-88fa-487a-bc64-f8ceb0708680.png)
5. Click on the "schedule a date" button and verify that you go directly to the Booking Form and don't see the Site Picker, i.e. that you navigate to `me/quickstart/SITE_iD/book`
6. Next, try navigating to `/purchases/subscriptions/SITE_WITHOUT_SESSION` and confirm the banner is not shown as there are no sessions available for this site. 
8. Finally, go to `/me/purchases` and verify the banner is shown there. Also, verify that clicking `Schedule a date` takes you directly to the Booking Form (the first site ID returned in the API data will be used in this case)

You can repeat steps 3-7 with a user that has paid sessions (see Phabricator Diff).

For more info on how to find more users with Quick Start sessions, take a look at the Test Plan at D86923-code.

**I don't know how to test users that have a pending appointment - if you're reviewing this and know how to test that, then please leave a comment**

For the pending appointment tests, you're going to need to use a mock/development back end API. You will also need to set the back end up to return an upcoming appointment or an available session. For that case, verify that you see the banner while you're using `en` as your locale. Switch your locale to another language (which you can do from your account settings), and when you navigate back to the two purchases pages, verify that you don't see the Quick Start banner.

#### Pre-merge Checklist



- [] Have you checked for TypeScript, React or other console errors?



Related to #67186
